Framework/spi

ATGr8 opened the pull request:
Wrote SPI wrapper and .h file

Commit:
Added SPI .c and .h file

Commit:
added error check

Commit:
added err check

Commit:
added parameters to functions in SPI wrapper

Commit:
idk what i did

Commit:
wrote SPI wrapper and header file

matthew-buglass commented:
Remember to add documentation to the C file.

Commit:
removed 'gu' I put there by mistake

Commit:
git p

ATGr8 commented:
> Remember to add documentation to the C file.

Thanks, I added them

matthew-buglass commented on .vscode/settings.json:
```diff
@@ -0,0 +1,7 @@
+{
+    "files.associations": {
```
I don't think that the IDE setting should be uploaded to the repo

matthew-buglass committed:
updated docs

I just specified explicitly that 1 was the failure code

matthew-buglass reviewed (changes requested):
Looks good, except for the .vscode settings being included. But I'm not sure what our policies on that, so I'll leave the to @ChrisAmaya to decide.

ChrisAmaya committed:
Deleting .json

Its usually never a good idea to commit your project settings 

Remember its bad practice to use "git add ." use "git add <filename>"

ChrisAmaya commented on radsat-sk/framework/wrappers/RSPI.h:
```diff
@@ -0,0 +1,23 @@
+#include <freertos/FreeRTOS.h>
+
+#define SPI_MAX_CS 8
+#define BUS 0
```
you have 2 defines but they arent being used anywhere in your .c files

ChrisAmaya commented on radsat-sk/framework/wrappers/RSPI.h:
```diff
@@ -0,0 +1,23 @@
+                                             FUNCTION DECLARATIONS
+***************************************************************************************************/
+
+uint32_t spiInit();
```
You have no parameters in your function declaration but you have then in your .c

Commit:
trying to get the include to work
